README.md

```diff
@@ -1,26 +1,242 @@
 # yarpc-go [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov]
 
-A transport and encoding agnostic RPC framework.
+With hundreds to thousands of services communicating with RPC, transport
+protocols (like HTTP and [TChannel][]), encoding protocols (like JSON or
+Thrift), and peer choosers are the concepts that vary.
+Separating these concerns allows services to change transport protocols without
+changing call sites or request handlers, build proxies and transport protocol
+bridges, or experiment with load balancing strategies.
+YARPC is a toolkit for services and proxies.
+
+[TChannel]: https://github.com/uber/tchannel
+
+YARPC breaks RPC into interchangeable encodings, transports, and peer
+choosers.
+YARPC for Go provides reference implementations for HTTP/1.1 and TChannel
+transports, and also raw, JSON, and Thrift encodings.
+YARPC for Go provides experimental implementations for a Redis transport and a
+round robin peer chooser.
+YARPC for Go plans to provide a Protobuf 3 encoding, a gRPC transport, and a
+load balancer that uses a least-pending-requests strategy.
+Peer choosers can implement any strategy, including load balancing or sharding,
+in turn backed by a pluggable peer provider.
+
+Regardless of transport, every RPC has some common properties: caller name,
+service name, procedure name, encoding name, deadline or TTL, headers, baggage
+(multi-hop headers), and tracing.
+Each RPC can also have an optional shard key, routing key, or routing delegate
+for advanced routing.
+YARPC transports use a shared API for capturing RPC metadata, so middleware can
+apply to requests over any transport.
+
+Each YARPC transport protocol can implement inbound handlers and outbound
+callers. Each of these can support different RPC types, like unary (request and
+response) or oneway (request and receipt) RPC. A future release of YARPC will
+add support for other RPC types including variations on streaming and pubsub.
+
 
 ## Installation
 
 ```
 go get -u go.uber.org/yarpc
 ```
 
-If using [Glide](https://github.com/Masterminds/glide), *at least* `glide version 0.12` is required to install:
+If using [Glide](https://github.com/Masterminds/glide), *at least* `glide
+version 0.12` is required to install:
 
 ```
 $ glide --version
 glide version 0.12.2
 
-$ glide get 'go.uber.org/yarpc#~0.5'
+$ glide get 'go.uber.org/yarpc#^1'
+```
+
+To use Thrift code generation, you will need to install plugins.
+These cannot be vendored since go depends on the binaries being available on
+the path.
+
+```
+$ go get 'go.uber.org/thriftrw'
+$ go get 'go.uber.org/yarpc/encoding/thrift/thriftrw-plugin-yarpc'
+```
+
+
+## Examples
```

Comment:
I would rather link directly to the example dirs, where we can add `README` to each one.